app/assets/javascripts/admin/order_cycles/controllers/order_cycle_exchanges_controller.js.coffee

```diff
@@ -37,5 +37,12 @@ angular.module('admin.orderCycles')
       $event.preventDefault()
       OrderCycle.addDistributor($scope.new_distributor_id)
 
+    $scope.setPickupTimeFieldDirty = (index) ->
+      setTimeout ->
+        last_outgoing_exchange_pickup_time_field_name = "order_cycle_outgoing_exchange_" + index + "_pickup_time"
+        $scope.order_cycle_form[last_outgoing_exchange_pickup_time_field_name].$setDirty()
+        $scope.$apply()
```

Comment:
Is `$scope.$apply()` definitely needed here?

Reply:
Update; if you use the `$timeout` service, you can remove `$scope.$apply()`.

Reply:
really nice one @Matt-Yorkley, apply was needed yes but with $timeout it's not needed. nice! new commit.